frontend/src/queries/query.ts

```diff
@@ -54,6 +55,12 @@ export function queryExportContext<N extends DataNode = DataNode>(
             currentTeamId: getCurrentTeamId(),
             refresh,
         })
+    } else if (isInsightVizNode(query)) {
+        return legacyInsightQueryExportContext({
+            filters: queryNodeToFilter(query.source),
+            currentTeamId: getCurrentTeamId(),
+            refresh,
+        })
```

Comment:
Could probably be done with the same recursive approach, as `DataTableNode` below i.e. something like this: 
```ts
return queryExportContext(query.source, methodOptions, refresh)
```

Reply:
🤦 way nicer! 💖